torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -138,13 +138,21 @@ bool FunctionParameter::check(PyObject* obj) {
       return THPVariable_Check(obj) || (allow_numbers_as_tensors && THPUtils_checkScalar(obj));
     }
     case ParameterType::SCALAR:
+      if (THPUtils_checkScalar(obj)) {
+        return true;
+      }
+      if (THPVariable_Check(obj)) {
+        auto & var = ((THPVariable*)obj)->cdata;
+        return !var.requires_grad() && var.dim() == 0;
+      }
+      return false;
     case ParameterType::COMPLEX:
       if (PyComplex_Check(obj)) {
         return true;
       }
       // fallthrough
     case ParameterType::DOUBLE: {
-      if (THPUtils_checkDouble(obj)) {
+      if (THPUtils_checkDouble(obj) || THPUtils_checkLong(obj)) {
```

Comment:
I think the problem was that `THPUtils_checkDouble` doesn't mean "check that the input is a double type", but rather, "check that the input is cast-able to double type".

Reply:
yes that ambiguity is part of the issue. I might be able to revert the meaning to the original 'castable' one after having fixed issues elsewhere though. I'll check.